draft-ietf-rtcweb-jsep.xml

```diff
@@ -576,6 +576,135 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
+          SDP attribute<xref target="RFC6236"/> to indicate:</t>
+
+          <t>
+            <list style="symbols">
+              <t>From the sender to the receiver: What video frame sizes it is
+              capable of sending, and what video frame sizes it prefers to
+              send</t>
+
+              <t>From the receiver to the sender: What video frame sizes it is
+              capable of receiving, and what video frame sizes it prefers to
+              receive</t>
+            </list>
+          </t>
+
+          <section title="Choice of imageattr recv attributes">
+            <t>The "a=imageattr" field is described as payload type specific.
+            When all video codecs supported have the same capabilities, use of
+            the wildcard payload type (*) is RECOMMENDED.</t>
+
+            <t>If video codecs have differing capabilities, the specific payload
+            types are used, and there is one "a=imageattr" line per payload
+            type.</t>
+
+            <t>The "a=imageattr" field can contain multiple sets of parameters,
+            each with a specified q value (a preference). In this specification,
+            the preferences are used in the following manner:</t>
+
+            <t><list style="symbols">
+              <t>q=0.1 is used to indicate the decode capabilities of the
+              device, if known. Using image sizes outside these ranges
+              risks decode failure.</t>
+
+              <t>q=0.5 is used to indicate the limits to image resolution
+              caused by mandatory constraints set by the user (if any).</t>
+
+              <t>q=1.0 is used to indicate the preferred format. Usually the
+              present format of the videostream will be preferred by the
+              sender and the size of the receiver's display surface will be
+              preferred by the receiver; if this format can be used, the video
+              stream can be passed through with an absolute minimum of
+              rescaling. If the user has specified "ideal" constraint values,
+              these might be used instead.</t>
+            </list>
+            Thus, a system with a HD decoder might send</t>
+
+            <t><list style="symbols">
+              <t>a=imageattr:* recv [x=[16:1920],y=[16:1080],q=0.1]</t>
+              </list>If the Javascript has constrained the stream to have
+            width={min:320, max:640}, the result might be</t>
+
+            <t><list style="symbols">
+              <t>a=imageattr:* recv [x=[16:1920],y=[16:1080],q=0.1]
+              [x=[320:640],y=[16:1080],q=0.5]</t>
+              </list>and if the stream is known to be displayed on a surface of
+            dimension 640x480, the result might be</t>
+
+            <t><list style="symbols">
+              <t>a=imageattr:* recv [x=[16:1920],y=[16:1080],q=0.1]
+              [x=[320:640],y=[16:1080],q=0.5] [x=640,y=480,q=1.0]</t>
```

Comment:
This breaks when the display resolution is larger than the constraints or decoder limits. Consider a decoder that does max 720p, and then the video element is made fullscreen (say, 1080p). Then, since q=1.0 for 1080p, the sender sends 1080p and the decoder dies.

Reversing the q values of course doesn't solve this problem, since if the display window is 480p, we don't want the sender to send 720p (recall, the decoder limit).

It's not clear to me how the sender should interpret multiple imageattrs with different q= values - i.e. should it try to satisfy the most preferred one, and then work backward, or should it try to satisfy them all? My interpretation has been the former, so I think that what we would need to do is to do some form of ANDing, e.g in the fullscreen/1080p case
q=0.1 is max 720p
q=1.0 is max 720p (1080p ANDed with 720p)

or in the 480p case
q=0.1 is max 720p
q=1.0 is max 480p (720p ANDed with 480p)